pkg/controller/core.oam.dev/v1alpha2/application/suite_test.go

```diff
@@ -166,3 +171,49 @@ var _ = AfterSuite(func() {
 	Expect(err).ToNot(HaveOccurred())
 	close(stop)
 })
+
+func collectEvents(source <-chan string) []string {
```

Comment:
I think we can even make the function as a method of `FakeRecorder`. And the fake recorder should record all events receved.

```
type FakeRecorder struct {
	Events chan string
        Message map[string][]string
}
```

We can filter all event messages for separated app name in message map.

And and a functon to get events by specified app.

Reply:
copy